frontend/public/components/events.jsx

```diff
@@ -83,207 +83,228 @@ const kindFilter = (reference, { involvedObject }) => {
+                      kind="Namespace"
+                      name={obj.namespace}
+                    />
+                  )}
+                  {lastTime && (
+                    <Timestamp className="co-sysevent__timestamp" timestamp={lastTime} />
                   )}
-                </small>
-                {count > 1 && (
-                  <small className="co-sysevent__count text-secondary">
-                    {count} times
-                    {firstTime && (
-                      <>
+                </div>
+                <div className="co-sysevent__details">
+                  <small className="co-sysevent__source">
+                    <Trans t={t} ns="events">
+                      Generated from <span>{{ sourceComponent: source.component }}</span>
+                    </Trans>
+                    {source.component === 'kubelet' && (
+                      <span>
                         {' '}
-                        in the last{' '}
-                        <Timestamp timestamp={firstTime} simple={true} omitSuffix={true} />
-                      </>
+                        {t('events~on ')}
+                        {flags[FLAGS.CAN_LIST_NODE] ? (
+                          <Link to={resourcePathFromModel(NodeModel, source.host)}>
+                            {source.host}
+                          </Link>
+                        ) : (
+                          <>{source.host}</>
+                        )}
+                      </span>
                     )}
                   </small>
-                )}
+                  {count > 1 && (
+                    <small className="co-sysevent__count text-secondary">
+                      {t('events~{{count}} times', { count })}
+                      {firstTime && (
+                        <>
+                          {' '}
+                          {t('events~in the last ')}
+                          <Timestamp timestamp={firstTime} simple={true} omitSuffix={true} />
+                        </>
+                      )}
+                    </small>
+                  )}
+                </div>
               </div>
+              <div className="co-sysevent__message">{message}</div>
             </div>
-
-            <div className="co-sysevent__message">{message}</div>
           </div>
-        </div>
-      );
-    }
-  },
+        );
+      }
+    },
+  ),
 );
 
-const eventTypes = { all: 'All Types', normal: 'Normal', warning: 'Warning' };
-
-export class EventsList extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      type: 'all',
-      textFilter: '',
-      selected: new Set(['All']),
-    };
-  }
+export const EventsList = (props) => {
+  const { t } = useTranslation();
+  const [type, setType] = React.useState('all');
+  const [textFilter, setTextFilter] = React.useState('');
+  const [selected, setSelected] = React.useState(new Set([t('events~All')]));
```

Comment:
We should be tracking some internal value here instead of saving the translated value in state. This might cause problems if the language is switched.

Reply:
added keyword all for all events instead of localized string